### PR TITLE
fix: error while starting trip entering odometer value and status while starting trip

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
fix: error while starting trip entering odometer value it was trying to update DN even if it does not exist in Delivery Stop and Delivery Trip status not being updated after starting the trip.
Ref: [Asana](https://app.asana.com/0/1198992588740629/1198881974636781)

Ref: [Asana](https://app.asana.com/0/1198992588740629/1198560613653269)

### CODE REVIEW

File: erpnext/stock/doctype/delivery_trip/delivery_trip.js [hunk 1]
```diff
@@ -155,8 +155,10 @@ frappe.ui.form.on('Delivery Trip', {
 										frm.reload_doc();
 									}
 								})
-								for(let stops of frm.doc.delivery_stops){
-									frappe.db.set_value("Delivery Note", stops.delivery_note, "status", "In Transit")
+								for(let stop of frm.doc.delivery_stops){
+									if(stop.delivery_note){
```
Comment (hrwX) [inline erpnext/stock/doctype/delivery_trip/delivery_trip.js]:
```suggestion
								for (let stop of frm.doc.delivery_stops) {
									if (stop.delivery_note) {
```

**Review by hrwX — CHANGES_REQUESTED** [review]

**Review by neilLasrado — DISMISSED** [review]

File: erpnext/stock/doctype/delivery_trip/delivery_trip.js [hunk 2]
```diff
@@ -155,8 +155,10 @@ frappe.ui.form.on('Delivery Trip', {
 										frm.reload_doc();
 									}
 								})
-								for(let stops of frm.doc.delivery_stops){
-									frappe.db.set_value("Delivery Note", stops.delivery_note, "status", "In Transit")
+								for (let stop of frm.doc.delivery_stops) {
+									if (stop.delivery_note) {
+									frappe.db.set_value("Delivery Note", stop.delivery_note, "status", "In Transit");
+									}
```
Comment (hrwX) [inline erpnext/stock/doctype/delivery_trip/delivery_trip.js]:
```suggestion
								for (let stop of frm.doc.delivery_stops) {
									if (stop.delivery_note) {
										frappe.db.set_value("Delivery Note", stop.delivery_note, "status", "In Transit");
									}
```

File: erpnext/stock/doctype/delivery_trip/delivery_trip.py [hunk 3]
```diff
@@ -87,7 +87,8 @@ def update_status(self):
 			elif any(visited_stops):
 				status = "In Transit"
 
-		self.status = status
+		if self.status != "In Transit" and self.status != "Completed":
+			self.status = status
```
Comment (hrwX) [inline erpnext/stock/doctype/delivery_trip/delivery_trip.py]:
```suggestion
```

File: erpnext/stock/doctype/delivery_trip/delivery_trip.py [hunk 4]
```diff
@@ -87,7 +87,8 @@ def update_status(self):
 			elif any(visited_stops):
```
Comment (hrwX) [inline erpnext/stock/doctype/delivery_trip/delivery_trip.py]:
```suggestion
			else:
```

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/delivery_trip/delivery_trip.js [hunk 5]
```diff
@@ -155,8 +155,10 @@ frappe.ui.form.on('Delivery Trip', {
 										frm.reload_doc();
 									}
 								})
-								for(let stops of frm.doc.delivery_stops){
-									frappe.db.set_value("Delivery Note", stops.delivery_note, "status", "In Transit")
+								for (let stop of frm.doc.delivery_stops) {
+									if (stop.delivery_note) {
+										frappe.db.set_value("Delivery Note", stop.delivery_note, "status", "In Transit");
+									}
```
Comment (rvpasari) [inline erpnext/stock/doctype/delivery_trip/delivery_trip.js:162]: Please put in a comment following javascript conventions explaining what this function call does. Will help better document our code

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**